### ISSUE TITLE
Failure! An error was encountered when trying to save your robots.txt content.

### ISSUE THREAD

**Issue by Deepa-Puranik** [GeekInTheNorth/Stott.Optimizely.RobotsHandler#56]
Hello,

When we try to update and save the robots.txt file, we are getting this error "Failure! An error was encountered when trying to save your robots.txt content." We have all the configuration as per the document.
Can you please help us what might be causing this error? 

Thanks,

Deepa

**Comment by GeekInTheNorth** [comment]
Hello @Deepa-Puranik 

Are you able to share an logs related to this event failure? They would contain any of the following three classes:

- RobotsContentService 
- RobotsContentRepository
- RobotsController

Also can you provide me with any JSON specific conventions you have set up for your solution as well as the CMS version you are having this issue on.

Thank You
Mark

**Comment by Deepa-Puranik** [comment]
Hello Mark,

CMS version is 12.22.6 and we have appsettings.json file in our solution.
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Information"
    }
  },
  "AllowedHosts": "*",
  "webpages:Version": "3.0.0.0",
  "webpages:Enabled": false,
  "PreserveLoginUrl": true,
  "ClientValidationEnabled": true,
  "UnobtrusiveJavaScriptEnabled": true,
  "ValidationSettings:UnobtrusiveValidationMode": "None",
  "aspnet:UseTaskFriendlySynchronizationContext": true,
  "MAICryptoKey": "X).Q7\u002Bc3c:=i}JGC.",
  "GetaEpiCategories:DisableCategoryAsLinkableType": true,
  "ImageProcessorDebug": true,
  "episerver:contentdelivery:maphttpattributeroutes": true,
  "episerver:contentdeliverysearch:maphttpattributeroutes": false,
  "Geta": {
    "NotFoundHandler": {
      "BufferSize": 40
    },
    "Sitemaps": {
      "EnableLanguageDropDownInAdmin": true
    }
  },
  "ConnectionStrings": {
    "EPiServerDB": "Replace in your own appsettings.Development.json file"
  },
  "EPiServer": {
    "Find": {
      "ServiceUrl": "Replace in your own appsettings.Development.json file",
      "DefaultIndex": "Replace in your own appsettings.Development.json file"
    },
    "CMS": {
      "LicensePath": {
        "Path": "Replace in your own appsettings.Development.json file"
      },
      "MappedRoles": {
        "Items": {
          "CmsEditors": {
            "MappedRoles": [ "WebEditors", "WebAdmins", "Contentteam", "Designteam", "Digitalteam", "Laserform", "ModernSystemsEditor", "News&Blogs" ]
          },
          "CmsAdmins": {
            "MappedRoles": [ "WebAdmins", "Administrators"]
          }
        }
      },
      "DataAccess": {
        "UpdateDatabaseSchema": "true"
      }
    }
  }
}
It's not logging any error in log file, just find out that we have custom rewriter in place and calling it using app.UseRewriter(RewriteOptionsHelper.rewriteOptions); If we remove this robots.txt works, we can't use rewriter in order to have this package working?


Thanks,
Deepa

**Comment by GeekInTheNorth** [comment]
Hello Deeper,

There must be something in the chosen rewriteOptions that is changing the URL so that the JS sees a 301/302 instead of a 200.

Can you tell me what your rewriteOptions are so I can reproduce and prevent this issue in future? alternatively can you update them to exclude the `/stott.robotshandler/` path? all routes for this module are on that path

Thanks,
Mark

**Comment by GeekInTheNorth** [comment]
Thank you @Deepa-Puranik I will try and replicate this issue and see if I can come to a solution for you.

**Comment by GeekInTheNorth** [comment]
@Deepa-Puranik The error was occurring because the save call was redirected to a new URL with a trailing slash.  The `/details/` API was already being called with a trailing slash so there was no redirect happening there, so I just corrected the `/save/` API to be called with a trailing slash.

SEO practice is typically to add a trailing slash rather than remove it.  So this will prevent those redirect rules firing.

I've uploaded 2.6.2 to nuget.org and api.nuget.optimizely.com to resolve this for you.
